tasks.py

```diff
@@ -242,7 +249,15 @@ def build(_c, debug=False):
             arg = command.upper().replace("-", "_") + "_VERSION"
             buildargs[arg] = CONFIG["commands"][command]["version"]
 
-    for variant in constants.VARIANTS:
+    variants = process_stages(stage=stage)
+
+    for variant in variants:
+        # This pulls the appropriate latest image from Docker Hub to be used as a cache when building
```

Comment:
After my cache testing (commits 3282c53 through 7e27d87 (see [this](https://github.com/SeisoLLC/easy_infra/pull/144/files/2666d69e637c5ce5affc9a14490300353f64b8de..7e27d87741ef558b748dca69ecbd7591229d2b6b) for the end result of those changes)), this appears to be by far the simplest (the cache is automatically updated for each commit to `main` by existing CI workflows) and most robust (works the same in CI and local, and is effective in both) without needing to manage cache hit keys or any additional update/restore logic.

Reply:
After spending a while reading about Github cache hits/misses and restore-key search priority, I'm somewhat surprised you were able to get such a robust solution with no key prefix management!